### PR TITLE
Replaced deprecated APIs and a bit more java8-ish

### PR DESCRIPTION
**Pull request by freesoft**

### CODE REVIEW

File: src/main/java/com/linecorp/armeria/server/ServiceMapping.java [hunk 1]
```diff
@@ -112,8 +112,7 @@ public MappedService apply(String path) {
         // Cache miss or disabled cache
         MappedService result = MappedService.empty();
         final int size = patterns.size();
-        for (int i = 0; i < size; i ++) {
-            final Entry<PathMapping, Service> e = patterns.get(i);
+        for (final Entry<PathMapping, Service> e : patterns) {
```
Comment (trustin) [inline src/main/java/com/linecorp/armeria/server/ServiceMapping.java]: The indexed loop here is intentional to avoid the instantiation of an Iterator. I.e. less garbage

**Comment by trustin** [comment]
Could you revert the file mode changes? 644 -> 755 isn't necessary here.

**Comment by freesoft** [comment]
I remember I did that because "sudo mvn clean package" failed in my Mac OS last night. Maven(actually maven ant plugin) wasn't able to create certain directory due to permission and it was resolve after my mode change. Let me try again later and will revert back if build can success without this.

File: src/main/java/com/linecorp/armeria/server/ServiceMapping.java (hunk 1 above)
Comment (freesoft) [inline src/main/java/com/linecorp/armeria/server/ServiceMapping.java]: Fair enough in case patterns arraylist is huge.

File: src/main/java/com/linecorp/armeria/client/http/ImmutableHttpHeaders.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
     @Override
     public String get(String name) {
-        return delegate.get(name);
+        return delegate.getAsString(name);
```
Comment (anuraaga) [inline src/main/java/com/linecorp/armeria/client/http/ImmutableHttpHeaders.java]: The HttpHeaders interface is currently quite messy, but this class is meant to be a simple proxy to HttpHeaders so it's better to just have it call the same method on the delegate.

Can you revert these and instead add this?
```
@SuppressWarnings("deprecation")
```

Comment (freesoft) [inline src/main/java/com/linecorp/armeria/client/http/ImmutableHttpHeaders.java]: I don't like the idea adding SupressWarnings. It should be either fixed to use replacement API or leave it as it is so that IDE users can know the function is using deprecated API.

Comment (anuraaga) [inline src/main/java/com/linecorp/armeria/client/http/ImmutableHttpHeaders.java]: SuppressWarnings should only apply to this method, not to users of the method (which there can't be since this class is not public).

I think it's intentional for this class to have deprecated methods that call deprecated methods on HttpHeaders since the intention is for it to be exactly the same, except for not exposting mutable methods. But we can clean up the warnings from the armeria build by suppressing here.

Comment (trustin) [inline src/main/java/com/linecorp/armeria/client/http/ImmutableHttpHeaders.java]: I find it delegating as is better because `ImmutableHttpHeaders` is just a wrapper class that is intended to delegate read-only operations of `HttpHeaders`. Please also note that `ImmutableHttpHeaders` is package-private and thus there's no chance of users' seeing it ever. No matter what you do, a user will see `HttpHeaders` and it's up to a user to avoid calling a deprecated method.

**Comment by trustin** [comment]
Thanks, but I ended up merging #9 instead, which includes the same fix and more.

**Comment by trustin** [comment]
.. and please don't get discouraged - let me look forward your upcoming contributions :-)
